### PR TITLE
Update JPA 3.1 FAT, update Eclipselink 4.0 with 4.0.0.3a0a517

### PR DESCRIPTION
**Pull request by jgrassel**
Updates Eclipselink to 4.0.0.3a0a517 and resolves ECL1544, ECL1625 and ECL1535

### CODE REVIEW

**Comment by jgrassel** [comment]
#build

File: dev/com.ibm.ws.jpa.tests.jpa_31_fat/test-applications/jpa31/src/io/openliberty/jpa/tests/jpa31/web/TestNewQueryTimeFunctionsServlet.java [hunk 1]
```diff
@@ -602,12 +602,12 @@ public void testEXTRACTFunction_OnQueryResults_JPQL() throws Exception {
...
-            result = q.getSingleResult();
-            Assert.assertNotNull(result);
-            Assert.assertEquals(0.0, result);
-        }
+//        if (!isSQLServer() && !isDB2ForLUW()) { // TODO, see https://github.com/eclipse-ee4j/eclipselink/issues/1573
+        q = em.createQuery("SELECT EXTRACT(SECOND FROM qdte.localDateTimeData) FROM QueryDateTimeEntity qdte WHERE qdte.id = 1");
+        result = q.getSingleResult();
+        Assert.assertNotNull(result);
+        Assert.assertEquals(0.0d, result);
+//        }
```
Comment (dazey3) [inline dev/com.ibm.ws.jpa.tests.jpa_31_fat/test-applications/jpa31/src/io/openliberty/jpa/tests/jpa31/web/TestNewQueryTimeFunctionsServlet.java]: Should these comments be removed now that the issues/TODOs are resolved?

File: dev/com.ibm.ws.jpa.tests.jpa_31_fat/test-applications/jpa31/src/io/openliberty/jpa/tests/jpa31/web/TestNewQueryTimeFunctionsServlet.java [hunk 2]
```diff
@@ -172,9 +172,9 @@ public void testLocalDateFunction_JPQL() throws Exception {
     @Test
     public void testLocalTimeFunction_JPQL() throws Exception {
         // TODO: Derby needs to switch from TIMESTAMP to TIME for this to work right; https://github.com/eclipse-ee4j/eclipselink/issues/1544
-        if (isDerby())
-            return;
-
+//        if (isDerby())
+//            return;
+//
```
Comment (dazey3) [inline dev/com.ibm.ws.jpa.tests.jpa_31_fat/test-applications/jpa31/src/io/openliberty/jpa/tests/jpa31/web/TestNewQueryTimeFunctionsServlet.java]: Should these comments be removed now that the issues/TODOs are resolved?

File: dev/com.ibm.ws.jpa.tests.jpa_31_fat/test-applications/jpa31/src/io/openliberty/jpa/tests/jpa31/web/TestNewQueryTimeFunctionsServlet.java (hunk 1 above)
Comment (jgrassel) [reply dev/com.ibm.ws.jpa.tests.jpa_31_fat/test-applications/jpa31/src/io/openliberty/jpa/tests/jpa31/web/TestNewQueryTimeFunctionsServlet.java]: Probably should, yeah.

File: dev/com.ibm.ws.jpa.tests.jpa_31_fat/test-applications/jpa31/src/io/openliberty/jpa/tests/jpa31/web/TestNewQueryTimeFunctionsServlet.java (hunk 2 above)
Comment (jgrassel) [reply dev/com.ibm.ws.jpa.tests.jpa_31_fat/test-applications/jpa31/src/io/openliberty/jpa/tests/jpa31/web/TestNewQueryTimeFunctionsServlet.java]: Probably should, yeah.

Comment (jgrassel) [reply dev/com.ibm.ws.jpa.tests.jpa_31_fat/test-applications/jpa31/src/io/openliberty/jpa/tests/jpa31/web/TestNewQueryTimeFunctionsServlet.java]: fixed

**Comment by jgrassel** [comment]
#run-libby-bot

**Review by dazey3 — APPROVED** [review]
LGTM

**Pull request merged**